ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java

```diff
@@ -143,6 +150,8 @@ public void startup() throws InstantiationException {
           networkServer.getRequestResponseChannel(), requests);
       networkServer.start();
 
+      clusterParticipant.initialize(networkConfig.hostName, networkConfig.port);
```

Comment:
Participant has a corresponding `Instance` in Helix. When participant disconnects, Helix considers the instance as down and ensures that some other instance is the LEADER for all the partitions for which the down instance was the LEADER. Going forward, when the Ambry server detects disk failures, it will simply update this information in its `InstanceConfig`.